src/Microsoft.ComponentDetection.Contracts/TypedComponent/VcpkgComponent.cs

```diff
@@ -42,14 +42,19 @@ public override string Id
     {
         get
         {
-            if (this.PortVersion > 0)
+            var componentLocationPrefix = string.Empty;
+            if (!string.IsNullOrWhiteSpace(this.DownloadLocation) && !this.DownloadLocation.Trim().Equals("NONE", System.StringComparison.InvariantCultureIgnoreCase))
             {
-                return $"{this.Name} {this.Version}#{this.PortVersion} - {this.Type}";
+                componentLocationPrefix = $"{this.DownloadLocation} : ";
             }
-            else
+
+            var componentPortVersionSuffix = " ";
+            if (this.PortVersion > 0)
             {
-                return $"{this.Name} {this.Version} - {this.Type}";
+                componentPortVersionSuffix = $"#{this.PortVersion} ";
             }
+
+            return $"{componentLocationPrefix}{this.Name} {this.Version}{componentPortVersionSuffix}- {this.Type}";
```

Comment:
do we not need space after `{componentPortVdrsionSuffix}` for portVersion > 0 scenario ? 

if yes, we can write it like this ...
```
 var componentPortVersionSuffix = EMTPTY;
            if (this.PortVersion > 0)
            {
                componentPortVersionSuffix = $"#{this.PortVersion} ";
            }

            return $"{componentLocationPrefix}{this.Name} {this.Version}{componentPortVersionSuffix} - {this.Type}";
```

Reply:
Yes that space is as suffix in line 54